src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -1600,7 +1600,7 @@ private async ValueTask<Http2Stream> SendHeadersAsync(HttpRequestMessage request
                 // Start the write.  This serializes access to write to the connection, and ensures that HEADERS
                 // and CONTINUATION frames stay together, as they must do. We use the lock as well to ensure new
                 // streams are created and started in order.
-                await PerformWriteAsync(totalSize, (thisRef: this, http2Stream, headerBytes, endStream: (request.Content == null && !http2Stream.ConnectProtocolEstablished), mustFlush), static (s, writeBuffer) =>
+                await PerformWriteAsync(totalSize, (thisRef: this, http2Stream, headerBytes, endStream: (request.Content == null && !request.IsWebSocketH2Request()), mustFlush), static (s, writeBuffer) =>
```

Comment:
I thought we were keeping websocket stuff out of httpclient?

Reply:
I guess we can add a property to request
```c# 
IsExtendedConnectRequest => Method == HttpMethod.Connect && _headers?.Protocol != null;
```
that should be enough for this check and most possibly for all other checks as well, so we could remove `IsWebSocketH2Request`?
Thoughts @greenEkatherine @stephentoub ?

Reply:
Why do you think it is not enough to use IsExtendedConnectRequest when receiving 200?

Reply:
I'm not sure that this status code would be correct for other protocols

Reply:
We've discussed offline that we would address this problem when it arises. Currently there's no evidence other protocols would use something else.